### PR TITLE
Str just funcs

### PR DESCRIPTION
**Pull request by tjhance**
Implement `str` functions `ljust`, `rjust`, and `center`.

### CODE REVIEW

File: src/runtime/str.cpp [hunk 1]
```diff
@@ -976,6 +976,57 @@ extern "C" Box* strNe(BoxedString* lhs, Box* rhs) {
...
+            padRight = 0;
+            break;
+        case JUST_CENTER:
+            padLeft = nNeeded / 2 + (nNeeded & targetWidth & 1);
+            padRight = nNeeded - padLeft;
+            break;
+    }
+
+    std::string res = std::string(padLeft, c) + self->s + std::string(padRight, c);
+    return new BoxedString(res);
```
Comment (undingen) [inline src/runtime/str.cpp]: nit-picking: you could add a ```std::move(res)``` here to avoid a copy.

**Comment by kmod** [comment]
The str functions look good, but did you mean to include the sys_test.py change?

I'm definitely curious as to why it's failing.  What does `make check_sys_test` (new convenience target) say?

**Comment by tjhance** [comment]
Sorry, I should have mentioned that...

Yeah it was failing on my machine. My sys.exec_prefix for CPython is different than Pyston's hardcoded value `/usr`:

```C++
sys_module->giveAttr("exec_prefix", boxStrConstant("/usr"));
```

Since Pyston doesn't really have a way to configure it, it seems like the functionality isn't really implemented and shouldn't be tested. I could change it to `# skip-if: sys.exec_prefix != '/usr'` or something though.

**Comment by kmod** [comment]
Well, this is something that should work (it gets used when self-host building extension modules).  Could we separate this into a separate PR?  It might not be too bad to do it the right way and port CPython's `Py_GetExecPrefix` and friends.

**Pull request merged**